<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>TestEditor</class>
 <widget class="QMainWindow" name="TestEditor">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>464</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Test Editor</string>
  </property>
  <property name="windowIcon">
   <iconset theme="QIcon::ThemeIcon::GoHome"/>
  </property>
  <property name="styleSheet">
   <string notr="true">/* ===== БАЗА: светлая тема ===== */
QMainWindow {
    background-color: #f5f7fb;
    font-family: &quot;Segoe UI&quot;;
    font-size: 10pt;
    color: #1f2937;
}

QWidget {
    color: #1f2937;
}

QWidget#questionBody QCheckBox::indicator {
	width: 16px;
    height: 16px;
    border: 1px solid #0b5ed7;
    background-color: white;
	border-radius: 4px;
}

QWidget#questionBody QCheckBox::indicator:checked {
	background-color: #0b5ed7;

}

QPushButton#addQuestion{
	background-color:#22c55e;
	color: white;
}

QPushButton#endTest {
	background-color: #0b5ed7;
	color: white;
}

QWidget#questionsPanel QPushButton{
	background-color: #ef4444;
	color: white;
}
/* ===== ЛЕВАЯ ПАНЕЛЬ ===== */
QWidget#sidePanel {
    background-color: #0b5ed7; 
	border-left: none;      /* синий */
    border-radius: 12px;

}
QLabel#qHeaderLabel{
	/*color: red;*/
	font-size: 25pt;
}
QLabel#lblMenu {
    color: white;
    font-size: 20pt;
    font-weight: 600;
}

QLabel#lblUserName,
QLabel#lblRole {
    color: rgba(255, 255, 255, 220);
    font-size: 10pt;
}

/* Кнопки меню */
QPushButton#btnTests,
QPushButton#btnStats,
QPushButton#btnLogout {
    background: transparent;
    color: white;
    border: none;
    text-align: center;
    padding: 10px 12px;
    border-radius: 8px;
    font-weight: 600;
    font-size: 12pt;

}
QWidget#questionsPanel
{
	background-color: #ffffff;
    border: 1px solid #e5e7eb;
    border-radius: 12px;
}
/* Hover */
QPushButton#btnTests:hover,
QPushButton#btnStats:hover,
QPushButton#btnLogout:hover {
    background-color: rgba(255, 255, 255, 35);
}

/* Pressed */
QPushButton#btnTests:pressed,
QPushButton#btnStats:pressed,
QPushButton#btnLogout:pressed {
    background-color: rgba(0, 0, 0, 25);
}

/* Чтобы можно было программно подсветить активную кнопку:
   ui-&gt;btnTests-&gt;setProperty(&quot;active&quot;, true); style()-&gt;polish(ui-&gt;btnTests);
*/
QPushButton[active=&quot;true&quot;] {
    background-color: rgba(255, 255, 255, 55);
    border-left: 4px solid #ffffff;
    padding-left: 8px; /* чтобы компенсировать border-left */
}

/* ===== ОБЛАСТЬ СТРАНИЦ (справа) ===== */
QStackedWidget {
    background: transparent;
}

/* ===== ПАНЕЛЬ ФИЛЬТРОВ ===== */
QWidget#filtersPanel {
    background-color: #ffffff;
    border: 1px solid #e5e7eb;
    border-radius: 12px;
}

QLabel#lblTests {
    font-size: 14pt;
    font-weight: 700;
    color: #111827;
}
QLabel#lblSearch, 
QLabel#lblSubject, 
QLabel#lblDifficult{
    font-size: 12pt;
    font-weight: 500;
    color: #111827;
}
/* Поля ввода и комбобоксы */
QLineEdit {
    background-color: #ffffff;
    border: 1px solid #d1d5db;
    border-radius: 8px;
    padding: 6px 10px;
    selection-background-color: #0b5ed7;
}

QLineEdit:focus {
    border: 1px solid #0b5ed7;
}

QComboBox {
    background-color: #ffffff;
    border: 1px solid #d1d5db;
    border-radius: 8px;
    padding: 6px 34px 6px 10px;  /* справа место под стрелку */
}

QComboBox::drop-down {
    subcontrol-origin: padding;
    subcontrol-position: top right;
    width: 28px;
    border-left: 1px solid #e5e7eb;
    border-top-right-radius: 8px;
    border-bottom-right-radius: 8px;
    background: transparent;
}

/* Рисуем V-стрелку только рамками */
QComboBox::down-arrow {
    width: 7px;
    height: 7px;
    border-left: 2px solid #0b5ed7;
    border-bottom: 2px solid #0b5ed7;
    margin-right: 1px;
}

/* Hover/focus */
QComboBox:hover {
    border: 1px solid #0b5ed7;
}

QComboBox::drop-down:hover {
    background-color: rgba(11, 94, 215, 10);
}


/* ===== ТАБЛИЦА ТЕСТОВ ===== */
QTableWidget {
    background-color: #ffffff;
    border: 1px solid #e5e7eb;
    border-radius: 12px;
    gridline-color: #eef2f7;
    selection-background-color: rgba(11, 94, 215, 35);
    selection-color: #111827;
}

/* Заголовок таблицы */
QHeaderView::section {
    background-color: #f3f6fb;
    color: #111827;
    padding: 8px 10px;
    border: none;
    border-bottom: 1px solid #e5e7eb;
    font-weight: 600;
}

/* Строки */
QTableWidget::item {
    padding: 6px 8px;
    border: none;
}

QTableWidget::item:hover {
    background-color: rgba(11, 94, 215, 18);
}

/* Скроллбар (аккуратный, синий акцент) */
QScrollBar:vertical {
    background: transparent;
    width: 10px;
    margin: 6px 4px 6px 0px;
}

QScrollBar::handle:vertical {
    background: rgba(11, 94, 215, 120);
    border-radius: 5px;
    min-height: 30px;
}

QScrollBar::handle:vertical:hover {
    background: rgba(11, 94, 215, 170);
}

QScrollBar::add-line:vertical,
QScrollBar::sub-line:vertical {
    height: 0px;
}

QScrollBar::add-page:vertical,
QScrollBar::sub-page:vertical {
    background: transparent;
}

/* ===== СТАТУС-БАР (если используешь) ===== */
QStatusBar {
    background-color: #ffffff;
    border-top: 1px solid #e5e7eb;
}

QWidget#handleTestsPanel{
	 background-color: #ffffff;
    border: 1px solid #e5e7eb;
    border-radius: 12px;
}
QPushButton#btnAddTest,
QPushButton#btnDeleteTest,
QPushButton#btnExportTest,
QPushButton#btnImportTest{
    background-color: #0b5ed7; 
    color: white;
	padding: 3px;
    border-radius: 8px;
    font-weight: 600;
    font-size: 10pt;

}
QPushButton#btnAddTest{
	background-color: rgb(34, 197, 94);
}
QPushButton#btnAddTest:hover{
	background-color: rgb(22, 163, 74);
}
QPushButton#btnDeleteTest{
	background-color: rgb(239, 68, 68);
}
QPushButton#btnDeleteTest:hover{
	background-color: rgb(220, 38, 38);
}
QPushButton#btnExportTest,
QPushButton#btnImportTest{
	background-color:rgb(249, 115, 22);
}
QPushButton#btnExportTest:hover,
QPushButton#btnImportTest:hover{
	background-color: rgb(234, 88, 12);
}</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="minimumSize">
    <size>
     <width>0</width>
     <height>0</height>
    </size>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout">
    <property name="leftMargin">
     <number>9</number>
    </property>
    <property name="topMargin">
     <number>9</number>
    </property>
    <property name="rightMargin">
     <number>9</number>
    </property>
    <property name="bottomMargin">
     <number>9</number>
    </property>
    <item>
     <widget class="QWidget" name="sidePanel" native="true">
      <property name="minimumSize">
       <size>
        <width>180</width>
        <height>0</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>220</width>
        <height>16777215</height>
       </size>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <property name="spacing">
        <number>8</number>
       </property>
       <property name="leftMargin">
        <number>9</number>
       </property>
       <property name="topMargin">
        <number>9</number>
       </property>
       <property name="rightMargin">
        <number>9</number>
       </property>
       <property name="bottomMargin">
        <number>9</number>
       </property>
       <item>
        <widget class="QLabel" name="lblMenu">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>60</height>
          </size>
         </property>
         <property name="text">
          <string>Меню</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignmentFlag::AlignCenter</set>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="btnTests">
         <property name="cursor">
          <cursorShape>PointingHandCursor</cursorShape>
         </property>
         <property name="text">
          <string>Тесты</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="btnStats">
         <property name="cursor">
          <cursorShape>PointingHandCursor</cursorShape>
         </property>
         <property name="text">
          <string>Статистика</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Orientation::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>200</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QLabel" name="lblUserName">
         <property name="text">
          <string>Имя пользователя</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="lblRole">
         <property name="text">
          <string>Роль</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="btnLogout">
         <property name="cursor">
          <cursorShape>PointingHandCursor</cursorShape>
         </property>
         <property name="text">
          <string>Выход</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QStackedWidget" name="stackedWidget">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>1</horstretch>
        <verstretch>1</verstretch>
       </sizepolicy>
      </property>
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="testsPage">
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="QWidget" name="filtersPanel" native="true">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>80</height>
           </size>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_3">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_4">
             <item>
              <widget class="QLabel" name="lblTests">
               <property name="text">
                <string>Редактор теста:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="edtTitle"/>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QWidget" name="horizontalWidget" native="true">
          <layout class="QHBoxLayout" name="bottomMenu">
           <item>
            <widget class="QPushButton" name="addQuestion">
             <property name="text">
              <string>Добавить вопрос</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="endTest">
             <property name="text">
              <string>Закончить тест</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="statisticsPage"/>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
